src/Team-Services-Bot.Api/Controllers/MessagesController.cs

```diff
@@ -25,16 +26,23 @@ public class MessagesController : ApiController
         /// </summary>
         public async Task<HttpResponseMessage> Post([FromBody]Activity activity)
         {
-            if (activity.Type == ActivityTypes.Message)
+            try
             {
-                await Conversation.SendAsync(activity, () => new Dialogs.RootDialog());
+                if (activity.Type == ActivityTypes.Message)
+                {
+                    await Conversation.SendAsync(activity, () => new Dialogs.RootDialog());
+                }
+                else
+                {
+                    HandleSystemMessage(activity);
+                }
+                var response = Request.CreateResponse(HttpStatusCode.OK);
+                return response;
             }
-            else
+            catch (Exception ex)
             {
-                HandleSystemMessage(activity);
+                throw ex;
```

Comment:
to re-throw an exception use "throw;" (ie don't quote the exception variable "ex").